cltk/tokenize/sentence.py

```diff
@@ -109,10 +133,114 @@ def indian_punctuation_tokenize_regex(self: object, untokenized_string: str):
+        """
         if self.language in INDIAN_LANGUAGES:
             return self.indian_punctuation_tokenize_regex(untokenized_string)
         else:
             return self.tokenize_sentences(untokenized_string)
+
+### Code below for consideration as new structure for modules; code above legacy?
+
+from abc import abstractmethod
+
+class BaseSentenceTokenizer(object):
+    """ Base class for sentence tokenization
+    """
+
+    def __init__(self: object, language: str = None):
+        """ Initialize stoplist builder with option for language specific parameters
+        :param language : language for sentence tokenization
+        :type language: str
+        """
+        if language:
+            self.language = language.lower()
+
+    @abstractmethod
+    def tokenize(self: object, text: str):
+        """
+        Method for tokenizing sentences. This method
+        should be overridden by subclasses of SentenceTokenizer.
+        """
+
+
+class BasePunktSentenceTokenizer(BaseSentenceTokenizer):
+    """Base class for punkt sentence tokenization
+    """
+
+    missing_models_message = "BasePunktSentenceTokenizer requires a language model."
+
+    def __init__(self: object, language: str = None, lang_vars: object = None):
+        """
+        :param language : language for sentence tokenization
+        :type language: str
+        """
+        self.language = language
+        self.lang_vars = lang_vars
+        BaseSentenceTokenizer.__init__(self, language=self.language)
+        # super().__init(language=self.language)
+        if self.language:
+            self.models_path = self._get_models_path(self.language)
+            try:
+                self.model =  open_pickle(os.path.join(os.path.expanduser(self.models_path), f'{self.language}_punkt.pickle'))
+            except FileNotFoundError as err:
+                raise type(err)(BasePunktSentenceTokenizer.missing_models_message)
+
+    def _get_models_path(self: object, language): # pragma: no cover
+        return f'~/cltk_data/{language}/model/{language}_models_cltk/tokenizers/sentence'
+
+    def tokenize(self: object, text: str, model: object = None):
+        """
+        Method for tokenizing sentences with pretrained punkt models; can
+        be overridden by language-specific tokenizers.
+
+        :rtype: list
+        :param text: text to be tokenized into sentences
+        :type text: str
+        :param model: tokenizer object to used # Should be in init?
+        :type model: object
+        """
+        if not self.model:
+            model = self.model
+
+        tokenizer = self.model
+        if self.lang_vars:
+            tokenizer._lang_vars = self.lang_vars
+
+        return tokenizer.tokenize(text)
+
+
+class BaseRegexSentenceTokenizer(BaseSentenceTokenizer):
+    """ Base class for regex sentence tokenization
+    """
+
+    def __init__(self: object, language: str = None, sent_end_chars: List[str] = []):
```

Comment:
This is actually a weird bug, you have to have None as a default argument and then assign a list.
See the correction at: https://gist.github.com/todd-cook/39027e4514df9edaa63875ec6be2864f